docs/ruleset_functions/get_spaces_associated_with_each_SWH_bat.md

```diff
@@ -0,0 +1,64 @@
+
+## get_spaces_associated_with_each_SWH_bat
+
+Description: This function gets all the spaces and the SHW space types and sorts them into a dictionary where the keys are the ServiceWaterHeatingSpaceOptions2019ASHRAE901 and values are a list if space.ids  
+
+Inputs:
+- **RMD**
+
+Returns:
+- **shw_and_spaces_dict**: A dictionary where the keys are the ServiceWaterHeatingSpaceOptions2019ASHRAE901 in the RMD and values are lists of space ids.  Example:  
+{"DORMITORY":["sp1","sp2","sp3"], "AUTOMOTIVE_FACILITY":["sp4","sp5","sp6"]}
+
+Function Call:
+
+- get_energy_required_to_heat_swh_use
+
+Data Lookup: None
+
+Logic:
+
+- create a blank dictionary: `shw_and_spaces_dict = {}`
+- look at each building segment in the RMD: `for building_segment in RMD...building_segments:`
+    - set the building_segment_shw_bat to nil: `building_segment_shw_bat = "UNDETERMINED"`
+    - if the building_segment.service_water_heating_area_type exists, include all of the spaces in this building segment in this area type: `if building_segment.service_water_heating_area_type:`
+        - set the building_segment_shw_bat to the building_segment.service_water_heating_area_type: `building_segment_shw_bat = building_segment.service_water_heating_area_type`
+
+    - otherwise, if the building segment doesn't have service_water_heating_area_type, we need to determine the type, we need to determine the building segment SHW type: `else:`
+        - create a dictionary that will hold the different types of swh_use_bat_types and the total service water used for the year: `swh_use_dict = {}`
+        - look at each space in the building segment: `for space in building_segment.spaces:`
+            - check to see if the space has service_water_heating_area_type: `if space.service_water_heating_area_type:`
+                - add the SWH building area type to the shw_use_dict and set the value to 0: `swh_use_dict.set_default(space.service_water_heating_area_type, 0)`
+                - look at each swh use for the space: `for swh_use in space.service_water_heating_uses:`
+                    - calculate the total hot water used using the function get_energy_required_to_heat_swh_use and add it to the shw_use_dict: `swh_use_dict[space.service_water_heating_area_type] += get_energy_required_to_heat_swh_use(swh_use)`
+            - otherwise, need to look at the individual swh uses: `else:`
```

Comment:
From the function logic it looks like you need to send both the shw_use and space to get_energy_required_to_heat_swh_use